crates/biome_configuration/src/javascript/mod.rs

```diff
@@ -50,3 +56,37 @@ pub struct JavascriptParser {
     #[partial(bpaf(hide))]
     pub unsafe_parameter_decorators_enabled: bool,
 }
+
+/// Indicates the type of runtime or transformation used for interpreting JSX.
+#[derive(
+    Bpaf, Clone, Copy, Debug, Default, Deserialize, Deserializable, Eq, Merge, PartialEq, Serialize,
+)]
+#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
+pub enum JsxRuntime {
+    /// Indicates a modern or native JSX environment, that doesn't require
+    /// special handling by Biome.
+    #[default]
+    Transparent,
+
+    /// Indicates a classic React environment that requires the `React` import.
+    ///
+    /// Corresponds to the `react` value for the `jsx` option in TypeScript's
+    /// `tsconfig.json`.
+    ///
+    /// This option should only be necessary if you cannot upgrade to a React
+    /// version that supports the new JSX runtime. For more information about
+    /// the old vs. new JSX runtime, please see:
+    /// https://legacy.reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html
+    ReactClassic,
+}
+
+impl FromStr for JsxRuntime {
+    type Err = String;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "transparent" => Ok(Self::Transparent),
+            "reactClassic" => Ok(Self::ReactClassic),
```

Comment:
Since this is a `bpaf` requirement, options via CLI should be dash-case. So `react-classic`